examples/functional-iteration/functional_iteration.bal

```diff
@@ -10,43 +10,58 @@ public function main() {
         e: "elephant"
     };
 
-    // The `count()` operation returns the number of elements in any collection.
-    io:println("Number of elements in 'words': ", words.count());
+    // The `.length()` operation returns the number of elements in any collection.
+    io:println("Number of elements in 'words': ", words.length());
 
-    // The `map()` operation applies the given function to each item of the iterable collection and returns a new
-    // iterable collection of the same length. The result of the map operation is assigned to a `string[]` as it returns a
-    // collection of string variables.
-    string[] animals = words.map(toUpper);
+    // The `.map()` operation applies the given function to each item of the iterable
+    /// collection and returns a new iterable collection of the same length.
+    // The result of the map operation is assigned to a new `map<string>` as it returns a
+    // a new map with modified values.
+    map<string> animals = words.entries().map(toUpper);
     io:println(animals);
 
     // Defines an `array` of `int` values.
     int[] numbers = [-5, -3, 2, 7, 12];
-    // The `filter()` operation returns a collection of all the elements that satisfy the input predicate function.
-    // The `average()` operation returns the average of a collection of `int` or `float`.
-    // The `max()` operation returns the maximum element of a collection.
-    // The `min()` operation returns the minimum element of a collection.
-    // The `sum()` operation returns the sum of all elements of a collection.
-    float avg = numbers.filter(function (int i) returns boolean {
+
+    // The `.filter()` operation returns a collection of all the elements that satisfy the
+    // input predicate function.
+    int[] positive = numbers.filter(function (int i) returns boolean {
                                   return i >= 0;
-                              }).average();
-    io:println("Average of positive numbers: ", avg);
+                              });
+
+    // The `.forEach()` function executes a provided function once for each array element
+    numbers.forEach(function(int i) {
+        io:println(i);
+    });
+
+    // The `.reduce()` function executes a reducer function on each element
+    // of the array resulting in a single output value.
+    int total = numbers.reduce(totalReducer, 0);
```

Comment:
we can get rid of the `totalReducer()` function altogether and just simply pass `(accum, val) => accum + val`